pyccel/parser/semantic.py

```diff
@@ -1132,6 +1136,73 @@ def _handle_function(self, expr, func, args, *, is_method = False, use_build_fun
                             func, func.is_elemental)
 
             return new_expr
+    def _handle_kernel(self, expr, func, args, **settings):
+        """
+        Create the node representing the kernel function call.
+
+        Create a FunctionCall or an instance of a PyccelInternalFunction
+        from the function information and arguments.
+
+        Parameters
+        ----------
+        expr : IndexedFunctionCall
+               Node has all the information about the function call
+
+        func : FunctionDef instance, Interface instance or PyccelInternalFunction type
+               The function being called.
+
+        args : tuple
+               The arguments passed to the function.
+
+        **settings : dict
+            The settings passed to _visit functions.
+
+        Returns
+        -------
+        Pyccel.ast.cuda.KernelCall
+            The semantic representation of the kernel call.
+        """
+        if(len(expr.launch_config) < 2):
+            errors.report(INVALID_KERNEL_LAUNCH_CONFIG_LOW,
+                    symbol=expr,
+                    severity='fatal')
+        if(len(expr.launch_config) > 2):
+            errors.report(INVALID_KERNEL_LAUNCH_CONFIG_HIGH,
+                    symbol=expr,
+                    severity='fatal')
+        if(len(func.results)):
+            errors.report(f"cuda kernel function '{func.name}' returned a value in violation of the laid-down specification",
+                         symbol=expr,
+                         severity='fatal')
+        if isinstance(func, FunctionDef) and len(args) != len(func.arguments):
+            errors.report(f"{len(args)} argument types given, but function takes {len(func.arguments)} arguments",
+                symbol=expr,
+                severity='fatal')
+        if not isinstance(expr.launch_config[0], (LiteralInteger, PythonTuple)):
```

Comment:
I'm surprised to see that `launch_config` can be a tuple. I would expect this to be unpacked earlier (at least before checking the length). Where is the unpacking done? Do you have a test for this?

Reply:
Sorry, I didn't understand. Could you please explain it in more detail?

Reply:
You check  `if not isinstance(expr.launch_config[0], (LiteralInteger, PythonTuple)):`. If I understood correctly, this means that you expect to be able to handle: `expr.launch_config = ((1,2),)`

Reply:
I don't see a test with:
```python
kernel[(1,2)]()
```
And I am not convinced that the code handles this correctly. In my example here `expr.launch_config[0]` is a `PythonTuple`

Reply:
I think I already changed that in Jalal's review

Reply:
So we decided that we are going to work with int for the moment 2cf6ed0

Reply:
Actually I think we only excluded tuples as arguments. But I'm not sure that this syntax `kernel[(1,2)]()
` really makes sense. Something like:
```python
a = (1,2)
kernel[a]()
```
might make sense, but `a` is not a `PythonTuple` here

Reply:
not a valid  syntax and the error that I raised makes sense in that case object of type 'int' has no len() it's the same as numba

Reply:
> not a valid syntax and the error that I raised makes sense in that case object of type 'int' has no len() it's the same as numba

This error makes no sense to me for 2 reasons:
1. I am not passing an int
2. `a` has a `len` operator

Reply:
can u check now please adad61a